File: src/main/java/com/freya02/botcommands/api/commands/annotations/BotPermissions.java
package com.freya02.botcommands.api.commands.annotations

import net.dv8tion.jda.api.Permission;

/**
 * Sets the required bot permissions to use this text / application command.
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION)
annotation class BotPermissions(vararg val value: Permission = [], val append: Boolean = false)

File: src/main/kotlin/io/github/freya022/botcommands/api/commands/annotations/BotPermissions.kt
package io.github.freya022.botcommands.api.commands.annotations

import net.dv8tion.jda.api.Permission

/**
 * Sets the required bot permissions to use this text / application command.
 */
@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
annotation class BotPermissions(@get:JvmName("value") vararg val permissions: Permission = [], val append: Boolean = false)
